File: CSE 241 Object Oriented Programming/2022 - 2023/HW3/driver1.cpp
#include <iostream>
#include "Tetris.h"
using namespace std;
using namespace GAME_TTS;

int main(){
    srand(time(nullptr));
    int row;
    int column;
    int key;
    //This part tests the Tetromino class
    cout << "-------------Testing Tetromino class----------"<<endl;
    cout << "Choose Tetromino Type"<<endl;
    cout << "\t1 - I"<< endl;
    cout << "\t2 - O"<< endl;
    cout << "\t3 - L"<< endl;
    cout << "\t4 - S"<< endl;
    cout << "\t5 - Z"<< endl;
    cout << "\t6 - T"<< endl;
    cout << "\t7 - J"<< endl;
    cout << "Choice: ";
    while((!(cin>>key)) || (key<1 || key>7)){
        cout << "Wrong choice: ";
        cin.clear();
        cin.ignore(10000, '\n');
    }
    Tetromino tetro(static_cast<TType>(key-1));
    cout << "Testing Tetromino \"print\" function."<<endl;    
    tetro.print();
    cout << "Testing Tetromino \"rotate\" function."<<endl;
    cout << "\tRotating right"<<endl;
    tetro.rotate(2);
    tetro.print();
    cout << "\tRotating left"<<endl;
    tetro.rotate(1);    
    tetro.print();

    
    //This part tests the Tetris class
    cout << "-------------Testing Tetris class----------"<<endl;
    cout << "Enter height : ";
    cin >> row;
    cout << "Enter width : ";
    cin >> column;
    cout << "Testing Constructor" << endl;
    Tetris game(row,column);
    cout << "Choose Tetromino Type"<<endl;
    cout << "\t1 - I"<< endl;
    cout << "\t2 - O"<< endl;
    cout << "\t3 - L"<< endl;
    cout << "\t4 - S"<< endl;
    cout << "\t5 - Z"<< endl;
    cout << "\t6 - T"<< endl;
    cout << "\t7 - J"<< endl;
    cout << "\t8 - Q(Quit)"<< endl;
    cout << "Choice: ";
    while((!(cin>>key)) || (key<=0 || key>8)){
        cout << "Wrong choice: ";
        cin.clear();
        cin.ignore(10000, '\n');
    }
    if(key == 8){
        exit(0);
    }
    Tetromino tetro2(static_cast<TType>(key-1));
    cout << "Adding tetromino to the board(Testing \"+=\" operator)"<<endl;
    game += tetro2;
    cout << "Testing Tetris \"draw\" function."<<endl;
    game.draw();
    cout << "Testing Tetris \"animate\" function."<<endl;
    game.animate();
}

File: CSE 241 Object Oriented Programming/2022 - 2023/HW7/src/driver1.java
import java.util.Scanner;
import java.util.Random;
/**
 * Driver for testing classes
 */
public class driver1 {
    public static void Wait(){
        try {
            Thread.sleep(500);
        } 
        catch (InterruptedException e) {
            Thread.currentThread().interrupt();
        }
    }
    public static void main(String args[]){
        int key;
        Scanner input = new Scanner(System.in);
        System.out.println("--------------------Testing Tetromino Class--------------------");
        System.out.println("-----------------Testing Tetromino Constructor-----------------");

        System.out.println("Choose Tetromino Type");
        System.out.println("\t1 - I");
        System.out.println("\t2 - O");
        System.out.println("\t3 - L");
        System.out.println("\t4 - S");
        System.out.println("\t5 - Z");
        System.out.println("\t6 - T");
        System.out.println("\t7 - J");
        do {
            System.out.print("Choice: ");
            while (!input.hasNextInt()) {
                System.out.print("Wrong Choice!!: ");
                input.next(); // this is important!
            }
            key = input.nextInt();
        } while (key <= 0 || key >7 || key<0);
        char tChoice = 'I';
        switch(key){
            case 1:
                tChoice = 'I';
                break;
            case 2:
                tChoice = 'O';
                break;
            case 3:
                tChoice = 'L';
                break;
            case 4:
                tChoice = 'S';
                break;                                        
            case 5:
                tChoice = 'Z';
                break;
            case 6:
                tChoice = 'T';
                break;
            case 7:
                tChoice = 'J';
                break;
        }
        Tetromino tetro = new Tetromino(tChoice);
        Wait();
        System.out.println("--------------------Testing Tetromino Print--------------------");
        tetro.print();
        Wait();
        System.out.println("-------------------Testing Tetromino Rotate--------------------");
        System.out.println("Rotate Left");
        tetro.rotate(0);
        tetro.print();
        Wait();
        System.out.println("Rotate Right");
        tetro.rotate(1);  
        tetro.print();
        Wait();
        System.out.println("----------------------Testing Tetris Class---------------------");
        System.out.println("------------------Testing Tetris Constructor-------------------");
        System.out.println("Creating 50x50 Tetris Board");
        Tetris game  = new Tetris(50, 50);
        Wait();
        System.out.println("Choose Tetromino Type");
        System.out.println("\t1 - I");
        System.out.println("\t2 - O");
        System.out.println("\t3 - L");
        System.out.println("\t4 - S");
        System.out.println("\t5 - Z");
        System.out.println("\t6 - T");
        System.out.println("\t7 - J");
        System.out.println("\t8 - Random");
        System.out.println("\t9 - Quit");
        do {
            System.out.print("Choice: ");
            while (!input.hasNextInt()) {
                System.out.print("Wrong Choice!!: ");
                input.next(); // this is important!
            }
            key = input.nextInt();
        } while (key <= 0 || key >9 || key<0);

        if(key == 8){
            Random rand = new Random();
            key = 1 + rand.nextInt(6);
        }
        if(key == 9){
           System.exit(0);
        }
        switch(key){
            case 1:
                tChoice = 'I';
                break;
            case 2:
                tChoice = 'O';
                break;
            case 3:
                tChoice = 'L';
                break;
            case 4:
                tChoice = 'S';
                break;                                        
            case 5:
                tChoice = 'Z';
                break;
            case 6:
                tChoice = 'T';
                break;
            case 7:
                tChoice = 'J';
                break;
        }
        tetro = new Tetromino(tChoice);
        Wait();
        System.out.println("----------------------Testing Tetris Add-----------------------");
        game.add(tetro);
        Wait();
        game.draw();
        System.out.println("----------------------Testing Tetris Draw----------------------");
        Wait();
        game.move(30, 5);
        game.draw();
        System.out.println("----------------------Testing Tetris Move----------------------");
        System.out.println("5 block downward and 5 block right");        
        System.out.println("Press any key to continue testing....");
        try{System.in.read();}
        catch(Exception e){}

        game.animate();
        System.out.println("--------------------Testing Tetris Animate---------------------");

        System.out.println("Press Enter to insert driver2");
        try{System.in.read();}
        catch(Exception e){}

        input.close();
    }
}
